twisted/internet/_sslverify.py

```diff
@@ -1645,11 +1648,11 @@ def _verifyCallback(conn, cert, errno, depth, preverify_ok):
             name = "%s-%d" % (reflect.qual(self.__class__), _sessionCounter())
             sessionName = md5(networkString(name)).hexdigest()
 
-            ctx.set_session_id(sessionName)
+            ctx.set_session_id(sessionName.encode('ascii'))
```

Comment:
On Python 2 this will allow users to pass in `unicode` objects that can be encoded with the `ascii` codec.  But the docstring for this method says that it wants a `list` of `bytes`.  I don't think the docstring should be changed; instead, I think you should add a helper method that ensures a given object is of type `bytes`, and optionally that ensures it contains only ASCII.  Here's what that might look like

```python
def isBytes(self, obj):
    if not isinstance(obj, bytes):
        raise TypeError("bytes required, not {!r}".format(obj))
    obj.decode('ascii')
    return obj
````

And in use:
```python
ctx.set_session_id(self.isBytes(sessionName))
```

I know this will be the second time you've been told to change your code :(  But implicit encoding concerns me when it's used in a security-sensitive API.